src/components/Select/Select.js

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { css, cx } from 'react-emotion';
+
+import { textMega } from '../../styles/style-helpers';
+import IconInputWrapper from '../IconInputWrapper';
+import DownTriangleIcon from './down-triangle.svg';
+
+const baseSelectStyles = ({ theme }) => css`
+  label: select;
+  appearance: none;
+  background-color: ${theme.colors.white};
+  border-width: 1px;
+  border-style: solid;
+  border-color: ${theme.colors.n300};
+  border-radius: ${theme.borderRadius.mega};
+  box-shadow: inset 0 1px 2px 0 rgba(102, 113, 123, 0.21);
+  color: ${theme.colors.n900};
+  padding: ${theme.spacings.byte} ${theme.spacings.tera} ${theme.spacings.byte}
+    ${theme.spacings.kilo};
+  position: relative;
+  ${textMega({ theme })};
+  z-index: 30;
+
+  &:focus,
+  &:hover,
+  &:active {
+    outline: none;
+  }
+
+  &:focus {
+    border: 1px solid ${theme.colors.b500};
+  }
+
+  &::after {
+  }
+
+  &:-moz-focusring {
+    color: transparent;
+    text-shadow: 0 0 0 #000;
+  }
+`;
+
+const disabledSelectStyles = ({ disabled }) =>
+  disabled &&
+  css`
+    label: select--disabled;
+    opacity: 0.4;
+    pointer-events: none;
+  `;
+
+const baseIconStyles = ({ theme }) => css`
+  label: select__icon;
+  fill: ${theme.colors.n700};
+  display: block;
+  z-index: 40;
+  pointer-events: none;
+`;
+
+const SelectElement = styled('select', { label: 'SelectElement' })(
+  baseSelectStyles,
+  disabledSelectStyles
+);
+const Icon = styled(DownTriangleIcon, { label: 'SelectIcon' })(baseIconStyles);
+
+/**
+ * A native select component.
+ */
+const Select = ({
+  options,
+  value,
+  placeholder,
+  selector,
+  disabled,
+  ...props
+}) => (
+  <IconInputWrapper
+    {...{ selector }}
+    iconPosition="right"
+    icon={({ className, disabledClassName }) => (
+      <Icon className={cx(className, { [disabledClassName]: disabled })} />
+    )}
+    input={({ className }) => (
+      <SelectElement {...{ ...props, className, disabled }}>
+        {!value && <option key={0}>{placeholder}</option>}
+        {options &&
+          options.map(option => (
+            <option key={option.value} {...option}>
+              {option.label}
+            </option>
+          ))}
+      </SelectElement>
+    )}
+  />
+);
+
+Select.propTypes = {
+  /**
+   * An ID rendered as data-selector attribute on the
+   * component. Used for tracking and e2e testing.
+   */
+  selector: PropTypes.string.isRequired,
+  /**
+   * onChange handler, called when the selection changes.
+   */
+  onChange: PropTypes.func.isRequired,
```

Comment:
A question - you're not using this prop explicitly, why define it and make it required? Shouldn't we only list the required props that we really need, and leave the rest to be defined by the user?

I mean, what if I don't want to handle my change event? It's not likely but there's no need to make this assumption, right?

Reply:
Yeah, point here was to impose certain restrictions so people would make less mistakes. But I guess, if the select does not work they will notice. Probably something we should discuss and put down somewhere as a guideline?